### PR TITLE
[pallet-revive] remove custom migration framework

### PR DESCRIPTION
**Pull request by pgherveou**
Fixes https://github.com/paritytech/polkadot-sdk/issues/5621

### CODE REVIEW

**Review by athei — APPROVED** [review]
Thanks

**Review by xermicus — APPROVED** [review]

**Comment by pgherveou** [comment]
Will merge that into #5608 once it has all approvals

**Pull request merged**
